### PR TITLE
(SLV-287) Add rake tasks and documentation to assist manual CD4PE setup

### PR DESCRIPTION
**Pull request by billclaytor**
The following rake tasks have been added to create the CD4PE test environment hosts via ABS:
```
rake abs_provision_environment_cd4pe          # Provision the hosts for a cd4pe test environment via ABS
rake abs_provision_host_agent                 # Provision an agent host via ABS
rake abs_provision_host_cd4pe                 # Provision a cd4pe host via ABS
rake abs_provision_host_gitlab                # Provision a gitlab host via ABS
rake abs_provision_host_worker                # Provision a worker host via ABS
```

The [CD4PE.md](CD4PE.md) file has been added to document the installation and configuration steps.

### CODE REVIEW

File: CD4PE.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+cd ssh && chown pe-puppet:pe-puppet id-control_repo.rsa id-control_repo.rsa.pub
+
+```
+
+* Get the public key
+```
+cat id-control_repo.rsa.pub
+```
+
+* Copy the key and add it to Gitlab
```
Comment (RandellP) [inline CD4PE.md]: could use a touch more detail on where in gitlab to add it.  I can't remember at the moment, but is it repo specific or general to gitlab... that kind of thing.

File: CD4PE.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+* Create a 'master' branch from the 'production' branch.
+* Set the 'master' branch as the default.
+* Create additional branches per environment.
+* Remove extraneous branches.
+* Ensure all branches are unprotected.
+
+# Set up PE
+## Enable Code Manager
+We'll be using code manager to deploy the cd4pe module, so we'll set it up before proceeding based on [the docs](https://puppet.com/docs/pe/2019.0/code_mgr_config.html#enable-code-manager-after-installation)
```
Comment (RandellP) [inline CD4PE.md]: When I read this i thought I was supposed to go to the doc and do what it says.  But really you are giving the link for reference or if the steps below don't work as expected...

File: CD4PE.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+* Get the public key
+```
+cat id-control_repo.rsa.pub
+```
+
+* Copy the key and add it to Gitlab
+
+### Enable Code Manager
+https://puppet.com/docs/pe/2019.0/code_mgr_config.html#enable-code-manager-after-installation
```
Comment (RandellP) [inline CD4PE.md]: same as above with this URL.  be sure to tell the reader this is just for reference...

File: CD4PE.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+### Import the control repo
+In the Gitlab UI:
+* Navigate to New Project → Import Project → Repo by URL
+* Provide a project name and description
+* Provide the control repo URL: https://github.com/puppetlabs/puppetlabs-puppetserver_perf_control.git
+* Click the 'Create project' button
+
+You should now see the control repo in Gitlab.
+* Disable the 'auto-dev-ops' pipeline
```
[the remark this answers is not in the document]
Comment (billclaytor) [reply CD4PE.md]: Updated

File: CD4PE.md (hunk 1 above)
Comment (billclaytor) [reply CD4PE.md]: Updated

File: CD4PE.md (hunk 2 above)
Comment (billclaytor) [reply CD4PE.md]: Updated

File: CD4PE.md (hunk 3 above)
Comment (billclaytor) [reply CD4PE.md]: Updated

**Pull request merged**